app/controllers/supervision/AnotherBodyController.scala

```diff
@@ -29,59 +29,66 @@ import views.html.supervision.another_body
 
 import scala.concurrent.Future
 
-class AnotherBodyController @Inject() (val dataCacheConnector: DataCacheConnector,
-                                       val authConnector: AuthConnector
-                                      ) extends BaseController {
+class AnotherBodyController @Inject()(val dataCacheConnector: DataCacheConnector,
+                                      val authConnector: AuthConnector
+                                     ) extends BaseController {
 
   def get(edit: Boolean = false) = Authorised.async {
-    implicit authContext => implicit request =>
-      dataCacheConnector.fetch[Supervision](Supervision.key) map {
-        response =>
-          val form: Form2[AnotherBody] = (for {
-            supervision <- response
-            anotherBody <- supervision.anotherBody
-          } yield Form2[AnotherBody](anotherBody)).getOrElse(EmptyForm)
-          Ok(another_body(form, edit))
-      }
+    implicit authContext =>
+      implicit request =>
+        dataCacheConnector.fetch[Supervision](Supervision.key) map {
+          response =>
+            val form: Form2[AnotherBody] = (for {
+              supervision <- response
+              anotherBody <- supervision.anotherBody
+            } yield Form2[AnotherBody](anotherBody)).getOrElse(EmptyForm)
+            Ok(another_body(form, edit))
+        }
   }
 
-  def post(edit : Boolean = false) = Authorised.async {
-    implicit authContext => implicit request =>
-      Form2[AnotherBody](request.body) match {
-        case f: InvalidForm =>
-          Future.successful(BadRequest(another_body(f, edit)))
-        case ValidForm(_, data: AnotherBody) =>
-          for {
-            supervision <- dataCacheConnector.fetch[Supervision](Supervision.key)
-            cache <- dataCacheConnector.save[Supervision](Supervision.key,
-              updateData(supervision, data))
-          } yield {
-            redirectTo(edit, cache)
-          }
-      }
+  def post(edit: Boolean = false) = Authorised.async {
+    implicit authContext =>
+      implicit request =>
+        Form2[AnotherBody](request.body) match {
+          case f: InvalidForm =>
+            Future.successful(BadRequest(another_body(f, edit)))
+          case ValidForm(_, data: AnotherBody) =>
+            for {
+              supervision <- dataCacheConnector.fetch[Supervision](Supervision.key)
+              _ <- dataCacheConnector.save[Supervision](Supervision.key, updateData(supervision, data))
+              maybeCache <- dataCacheConnector.fetchAll
+              cache <- Future.successful(maybeCache)
```

Comment:
As above. Should we map over here instead of calling future.successful? There are examples in the code where we map over a cache fetch.